### PR TITLE
Open Graph: add image:width and image:height when available

### LINKED ISSUE #2104: Open Graph: add og:image:width and og:image:height Open Graph tags
These 2 tags seem useful when pre-caching the image to be used in post previews:
https://developers.facebook.com/docs/sharing/best-practices#precaching

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #2104

- Grab the dimensions from `Jetpack_PostImages::get_images` when available
- Add dimensions to Site Logo and Site Icon fallbacks

### CODE REVIEW

File: modules/theme-tools/site-logo/inc/functions.php [hunk 1]
```diff
@@ -31,6 +31,50 @@ function jetpack_get_site_logo( $show = 'url' ) {
...
+ *
+ * @since 3.6.0
+ *
+ * @return array $dimensions {
+ *		An array of dimensions of the Site Logo.
+ *
+ * 		@type string $width Width of the logo in pixels.
+ * 		@type string $height Height of the logo in pixels.
+ * }
+}
```
Comment (dereksmart) [inline modules/theme-tools/site-logo/inc/functions.php]: looks like there's a stray brace

Comment (jeherve) [inline modules/theme-tools/site-logo/inc/functions.php]: Oops, how did I miss this? :( Removed in 987894f7071f4756a799b3029a4404f5620dc3d6

File: functions.opengraph.php [hunk 2]
```diff
@@ -97,8 +97,18 @@ function jetpack_og_tags() {
 
 	$tags['og:site_name'] = get_bloginfo( 'name' );
 
-	if ( !post_password_required() )
-		$tags['og:image']     = jetpack_og_get_image( $image_width, $image_height );
+	// Get image info and build tags
+	if ( ! post_password_required() ) {
+		$image_info                  = jetpack_og_get_image( $image_width, $image_height );
```
Comment (dereksmart) [inline functions.opengraph.php]: Why so much space?

File: functions.opengraph.php [hunk 3]
```diff
@@ -184,50 +198,64 @@ function jetpack_og_get_image( $width = 200, $height = 200, $max_images = 4 ) {
...
+			$image['height'] = $height;
+		}
 	}
 
 	// Second fall back, Site Logo
 	if ( empty( $image ) && ( function_exists( 'jetpack_has_site_logo' ) && jetpack_has_site_logo() ) ) {
-		$image[] = jetpack_get_site_logo( 'url' );
+		$image['src']        = jetpack_get_site_logo( 'url' );
+
+		$image_dimensions    = jetpack_get_site_logo_dimensions();
```
Comment (dereksmart) [inline functions.opengraph.php]: We would need to find a place to add `jetpack_get_site_logo_dimensions()` in wpcom

**Comment by dereksmart** [comment]
Going to punt for for now, some additional thought needs to go in to handle the changes in wpcom

**Comment by jeherve** [comment]
> Going to punt for for now, some additional thought needs to go in to handle the changes in wpcom

How about just adding 9961391e6f982023a8a20bc1397878afdbc53c0e in 3.6? This would already reduce our support volume a bit.

**Comment by dereksmart** [comment]
That WFM if it helps decrease support load!

**Comment by dereksmart** [comment]
I cherry-picked and merged half of this (9961391) with #2291.  Closing this, as it should be re-opened with just the other half of the changes.

**Comment by jeherve** [comment]
> Closing this, as it should be re-opened with just the other half of the changes.

I opened #2294 for the other half of the PR.

**Pull request closed without merging**